fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryManager.java

```diff
@@ -121,18 +129,58 @@ public CompletableFuture<Void> mergeIndex(LucenePartitioner partitioner, LuceneA
         return cursor
                 .map(tuple ->
                         Tuple.fromItems(tuple.getItems().subList(0, groupingCount)))
-                .forEach(groupingKey ->
-                        mergeIndex(analyzerWrapper, groupingKey, partitioner.isPartitioningEnabled() ? 0 : null));
+                .forEachAsync(groupingKey -> mergeIndex(analyzerWrapper, groupingKey, partitioner, agilityContext),
+                        2);
     }
 
-    private void mergeIndex(LuceneAnalyzerWrapper analyzerWrapper, Tuple groupingKey, @Nullable Integer partitionId) {
-        try {
-            getDirectoryWrapper(groupingKey, partitionId, true).mergeIndex(analyzerWrapper);
-        } catch (IOException e) {
-            throw new RecordCoreStorageException("Lucene mergeIndex failed", e);
+    private CompletableFuture<Void> mergeIndex(LuceneAnalyzerWrapper analyzerWrapper, Tuple groupingKey, @Nullable LucenePartitioner partitioner, final AgilityContext agileContext) {
+        if (!partitioner.isPartitioningEnabled()) {
```

Comment:
`partitioner` can be `null`

Reply:
It actually can't, I'll update the annotations.